# C0123 part:

In [159]:
import pandas as pd

In [160]:
#!pip3 install xlrd - uncomment if not installed
newc0123=pd.ExcelFile('K2 (C0-1-2-3) - RR Lyrae Signals (Update.v.1).xlsx')

In [161]:
newc0123.sheet_names # check sheet name

['Signals']

In [162]:
newc0123=newc0123.parse('Signals') # read in sheet

In [163]:
newc0123=newc0123[newc0123.columns[0]].tolist() # make a list from the first col of the sheet

In [164]:
newc0123=set(newc0123) # create a set from list

In [165]:
ownc0=pd.read_csv('c0.list',sep=' ',header=None)[0].tolist() #read in own c0 list, make it a list
ownc0=set([int(each[5:14]) for each in ownc0]) #keep only the ID and make it a set

In [166]:
ownc1=pd.read_csv('c1.list',sep=' ',header=None)[0].tolist() #read in own c1 l
ownc1=set([int(each[5:14]) for each in ownc1]) #keep only the ID and make it a set

In [167]:
ownc2=set() 
with open('C2') as ownc2in: #open own
    for each, line in enumerate(ownc2in):
        ownc2.add(int(line.split()[-1])) # read in each line, where there are two IDs, take the second one

In [168]:
import glob as g

In [169]:
ownc3=set([int(each [5:14]) for each in g.glob("EPIC*c03*")])#read in own c3 files, take only the ID, make it a set

In [170]:
c0123difference = newc0123 - ownc3 - ownc2 - ownc1 - ownc0 # check difference
c0123difference

{210282474, 'Star ID (EPIC)', nan}

# we dont have 210282474 on our list


# c4 part:

In [171]:
newc4=pd.ExcelFile('K2 (C4) - RR Lyrae Signals.xlsx')

In [172]:
newc4=newc4.parse('Signals') # read in sheet

In [173]:
newc4=set(newc4[newc4.columns[0]].tolist()) # make a list then set from the first col of the sheet

In [174]:
ownc4=set([int(each[5:14]) for each in g.glob('E*c04*')]) # get list of own c4, slice for ID, make it a set

In [175]:
c4difference = newc4-ownc4
c4difference

{210619926, 211062662, 211063000, 211069629, 'Star ID (EPIC)', nan}

# 210619926, 211062662, 211063000, 211069629 are new to us

## c6 part:

In [176]:
newc6=pd.read_csv('K2 C6 - RR Lyrae Candidates.txt') # read in file

In [177]:
newc6=set(newc6[newc6.columns[0]].tolist()) # get IDs from first column of file, make it a list, then set

In [178]:
ownc6=set([int(each[5:14]) for each in g.glob('E*c06*')])

In [179]:
c6difference = newc6-ownc6
c6difference

{212347134,
 212405442,
 212626187,
 212643422,
 212643647,
 212698817,
 212753920,
 229227274,
 229228121}

# 212347134, 212405442, 212626187, 212643422, 212643647, 212698817, 212753920, 229227274, 229228121 are new to us

# check if they RRLs:

In [229]:
wedidntfind0 = c0123difference | c4difference | c6difference 

In [248]:
wedidntfind0=list(wedidntfind)
wedidntfind0

[212753920,
 212698817,
 212405442,
 211062662,
 229227274,
 212626187,
 210619926,
 211063000,
 229228121,
 212643422,
 210282474,
 211069629,
 212347134,
 212643647]

In [249]:
wedidntfind=[]
for each in wedidntfind0:
    if (type(each) is int) == True: # we want to keep only the numerical values
        wedidntfind.append(each)

In [250]:
fout=open('wedidntfind','w')
for each in wedidntfind: fout.write(str(each)+'\n') # write out results to file
fout.close()

# use the file here: https://archive.stsci.edu/k2/data_search/search.php

https://archive.stsci.edu/canvas/k2hlsp_plot.html?k2=210282474&c=1&type=LC
https://archive.stsci.edu/canvas/k2hlsp_plot.html?k2=210619926&c=4&type=LC
https://archive.stsci.edu/canvas/k2hlsp_plot.html?k2=210619926&c=4&type=SC
https://archive.stsci.edu/canvas/k2hlsp_plot.html?k2=211062662&c=4&type=LC
https://archive.stsci.edu/canvas/k2hlsp_plot.html?k2=211062662&c=4&type=SC
https://archive.stsci.edu/canvas/k2hlsp_plot.html?k2=211063000&c=4&type=LC
https://archive.stsci.edu/canvas/k2hlsp_plot.html?k2=211069629&c=4&type=LC
https://archive.stsci.edu/canvas/k2hlsp_plot.html?k2=211069629&c=4&type=SC
https://archive.stsci.edu/canvas/k2hlsp_plot.html?k2=212347134&c=6&type=LC
https://archive.stsci.edu/canvas/k2hlsp_plot.html?k2=212405442&c=6&type=LC
https://archive.stsci.edu/canvas/k2hlsp_plot.html?k2=212626187&c=6&type=LC
https://archive.stsci.edu/canvas/k2hlsp_plot.html?k2=212643422&c=6&type=LC
https://archive.stsci.edu/canvas/k2hlsp_plot.html?k2=212643647&c=6&type=LC
https://archive.stsci.edu/canvas/k2hlsp_plot.html?k2=212698817&c=6&type=LC
https://archive.stsci.edu/canvas/k2hlsp_plot.html?k2=212753920&c=6&type=LC
https://archive.stsci.edu/canvas/k2hlsp_plot.html?k2=229227274&c=6&type=LC
https://archive.stsci.edu/canvas/k2hlsp_plot.html?k2=229228121&c=6&type=LC